packages/components/src/text/types.ts

```diff
@@ -35,7 +28,12 @@ export interface Props extends TruncateProps {
 	/**
 	 * Automatically calculate the appropriate line-height value for contents that render text and Control elements (e.g. `TextInput`).
 	 */
-	adjustLineHeightForInnerControls?: TextAdjustLineHeightForInnerControls;
+	adjustLineHeightForInnerControls?:
+		| boolean
+		| 'large'
+		| 'medium'
+		| 'small'
+		| 'xSmall';
```

Comment:
Maybe the "limit" for expanding type aliases is if the types that are unioned are all of the same general type or not?

E.g. a union of just strings could be "easier" to expand for the docgen, while a union of strings and boolean may be kept un-expanded?

Reply:
I don't know 🤷

Reply:
Me neither, but good to keep at the back of our minds for future work :)